experimental/Geometry/Misc.jl

```diff
@@ -0,0 +1,202 @@
+import AbstractAlgebra.Ring
+import AbstractAlgebra.RingElement
+import Oscar: radical_membership
+
+
+module Misc
+using Oscar
+
+export add_variables, divides_power
+
+####################################################################
+#  
+#  Miscellaneous routines used in the above 
+#
+####################################################################
+
+
+function add_variables( R::T, new_vars::Vector{String} ) where{ T<:MPolyRing } 
+  k = base_ring(R)
+  old_vars = String.( symbols(R) )
+  n = length( old_vars )
+  vars = vcat( old_vars, new_vars )
+  S, v = PolynomialRing( k, vars )
+  phi = AlgebraHomomorphism( R, S, gens(S)[1:n] )
+  y = v[n+1:length(v)]
+  return S, phi, y
+end
+
+function add_variables( P::T, new_vars::Vector{String} ) where{ T<:MPolyQuo } 
+  R = base_ring(P)
+  k = base_ring(R)
+  S, phi, y = add_variables( R, new_vars )
+  I = ideal( S, [ phi(g) for g in gens(modulus(P)) ] )
+  Q, projection = quo( S, I )
+  phi_quot = AlgebraHomomorphism( P, Q, [projection(g) for g in phi.image] )
+  return Q, phi_quot, [ projection(g) for g in y]
+end
+
+####################################################################
+#
+# Test whether some power of the polynomial f is contained in the 
+# principal ideal generated by g
+#
+function Oscar.radical_membership( f::U, g::U ) where{ U<:MPolyElem }
+  R = parent(f)
+  if R != parent(g) 
+    error( "Polynomials do not belong to the same ring" )
+  end
+  S, phi, t = add_variables( R, ["t"] )
+  I = ideal(S, [phi(g), one(S)-t[1]*phi(f)])
+  G = groebner_basis(I,complete_reduction=true)
+  if length(G)==1 && G[1] == one(S)
+    return true
+  end
+  return false
```

Comment:
Can't this be simplified to this? (Or do we loose performance somehow? But if so, shouldn't we improve the `ideal_membership` method?)
```suggestion
  return one(S) in I
```

Reply:
Good point. I think in principal we should rely on the ideal membership being implemented in more efficient ways, than oneself could possibly do it on the fly. So the suggested change seems correct. 

However, I would like to first do some performance tests with different implementations of this routine: Groebner-based vs. divides-based, etc. and then see what we really want to use. So we might switch to using the coefficients of the Groebner-basis methods in the end...

Btw: Did anyone yet think of providing ideal membership together with the coefficients writing the element as linear combination of the chosen generators? That would be nice to have, but again tend to contradict the general Oscar approach to hide away the algorithmic backside (as also discussed on Slack).

Reply:
Representing an element in terms of the generators canbe done by liftstd (from Singular.jl),
but is much more expensive than just a membership test:
Membership test can stop as soon as the ledading term cannot be reduced further,
liftstd requires a full reduction AND keeping track of all operations